### PR TITLE
Migrate ``rustc_middle`` diagnostic

### PR DESCRIPTION
**Pull request by MingyuChen1**
Part of #100717

### CODE REVIEW

File: compiler/rustc_error_messages/locales/en-US/middle.ftl [hunk 1]
```diff
@@ -0,0 +1,17 @@
+middle_drop_check_overflow =
+    overflow while adding drop-check rules for {$ty}
+    .note = {$note}
```
Comment (davidtwco) [inline compiler/rustc_error_messages/locales/en-US/middle.ftl:3]:
```suggestion
    .note = overflowed on {$overflow_ty}
```

Why not this?

**Review by davidtwco — COMMENTED** [review]
Thanks for working on this, I've left one comment with what you've got so far, I assume you're still working your way through the crate and not looking to merge yet?

File: compiler/rustc_error_messages/locales/en-US/middle.ftl (hunk 1 above)
Comment (MingyuChen1) [reply compiler/rustc_error_messages/locales/en-US/middle.ftl]: Sorry David, I haven't got much time to work on it lately. You can merge this one and unassign me from this crate. I will come back and pick up anything left when I got time.

**Review by davidtwco — APPROVED** [review]

**Pull request merged**